libvast/vast/system/node.hpp

```diff
@@ -27,6 +27,29 @@
 
 namespace vast::system {
 
+struct node_state;
+
+using node_actor = caf::stateful_actor<node_state>;
+
+struct node {
+
+  enum flags {
+    none = 0x0,
+    accountant = 0x1
+  };
+
+  /// Spawns a node.
+  /// @param self The actor handle
+  /// @param id The unique ID of the node.
+  /// @param dir The directory where to store persistent state.
+  /// @param node_flags Additional flags to customize behavior.
+  caf::behavior operator()(node_actor* self, std::string id, path dir,
+                           flags flags = flags::accountant);
+
+};
+
+extern node node;
+
```

Comment:
While I appreciate the beauty of this "workaround", it feels too heavyweight for the task at hand. You only want to avoid a boolean function parameter to make the accountant optional. The thinking is noble, but there might be a less bizarre construction to achieve the same.

Per our discussion, I think we can shoot down the accountant after spawning a `node` in the unit tests. I'd say just revert this very commit and start over.